Rocket.Chat/Models/Auth.swift

```diff
@@ -46,3 +51,42 @@ final class Auth: Object {
         return "serverURL"
     }
 }
+
+extension Auth {
+    enum CanDeleteMessageResult {
+        case allowed
+        case timeElapsed
+        case differentUser
+        case serverBlocked
+        case notActionable
+        case unknown
+    }
+
+    func canDeleteMessage(_ message: Message) -> CanDeleteMessageResult {
+        guard
+            let createdAt = message.createdAt,
+            let user = user,
+            let settings = settings
+        else {
+            return .unknown
+        }
+
+        guard message.type.actionable else { return .notActionable }
+
+        if user.hasPermission(.forceDeleteMessage, realm: self.realm) { return .allowed }
```

Comment:
Why using `if` and `guard` before here? What about using two `ifs`? I would also break the link on return to make easier to read the return type.